ssl/statem/extensions.c

```diff
@@ -939,9 +939,15 @@ static int final_server_name(SSL *s, unsigned int context, int sent)
      */
     if (s->server) {
         if (!sent) {
-            /* Nothing from the client this handshake; cleanup stale value */
-            OPENSSL_free(s->ext.hostname);
-            s->ext.hostname = NULL;
+            /*
+             * Nothing used from the client this handshake; from the
+             * perspective of TLS any value here is stale/unused.  However,
+             * we have a de facto stable ABI guarantee to preserve this value
+             * for subsequent retrieval with SSL_get_servername() even if it
+             * was not used to negotiate an SNI value for this session.
+             * TODO(OpenSSL1.2) cleanup stale value.
+             */
+            ;
```

Comment:
Is that semi necessary?  It strikes me as odd, but if there's a compiler complaining when it's not there, I can understand...

Reply:
I think I have seen a compiler that complains without the semicolon, but that would have been on the OpenAFS codebase and not the OpenSSL one.  I'm happy to push another change and see if the CI lets if by...